src/test/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/personalrelationships/profiledetails/ContactPersonProfileDetailsMigrationIntTest.kt

```diff
@@ -0,0 +1,197 @@
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.persistence.repository.MigrationHistoryRepository
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.wiremock.NomisApiExtension.Companion.nomisApi
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.wiremock.withRequestBodyJsonPath
+import uk.gov.justice.hmpps.test.kotlin.auth.WithMockAuthUser
+import java.time.Duration
+import java.time.LocalDate
+
+@WithMockAuthUser
+class ContactPersonProfileDetailsMigrationIntTest(
+  @Autowired private val nomisProfileDetailsApi: ContactPersonProfileDetailsNomisApiMockServer,
+  @Autowired private val mappingApi: ContactPersonProfileDetailsMappingApiMockServer,
+  @Autowired private val dpsApi: ContactPersonProfileDetailsDpsApiMockServer,
+  @Autowired private val migrationService: ContactPersonProfileDetailsMigrationService,
+  @Autowired private val migrationHistoryRepository: MigrationHistoryRepository,
+) : SqsIntegrationTestBase() {
+
+  @Nested
+  inner class Migrate {
+    private lateinit var migrationId: String
+
+    @BeforeEach
+    internal fun deleteHistoryRecords() {
+      runBlocking {
+        migrationHistoryRepository.deleteAll()
+      }
+    }
+
+    private fun stubMigrationDependencies(entities: Int = 2) {
+      nomisApi.stubGetPrisonerIds(totalElements = entities.toLong(), pageSize = 10, firstOffenderNo = "A0001KT")
+      (1L..entities)
+        .map { "A000${it}KT" }
+        .forEachIndexed { index, offenderNo ->
+          nomisProfileDetailsApi.stubGetProfileDetails(
+            offenderNo = offenderNo,
+            bookingId = null,
+            response = profileDetailsResponse(
+              offenderNo,
+              listOf(
+                booking(
+                  bookingId = index.toLong(),
+                  latestBooking = true,
+                  listOf(
+                    profileDetails("MARITAL", "${10 * index + 1}"),
+                    profileDetails("CHILD", "${10 * index + 2}"),
+                  ),
+                ),
+                booking(
+                  bookingId = 10 + index.toLong(),
+                  latestBooking = false,
+                  listOf(
+                    profileDetails("MARITAL", "${10 * index + 3}"),
+                    profileDetails("CHILD", "${10 * index + 4}"),
+                  ),
+                ),
+              ),
+            ),
+          )
+          dpsApi.stubMigrateDomesticStatus(migrateDomesticStatusResponse(offenderNo))
+          dpsApi.stubMigrateNumberOfChildren(migrateNumberOfChildrenResponse(offenderNo))
+          mappingApi.stubPutMapping()
+        }
+    }
+
+    @Nested
+    inner class HappyPath {
+      @BeforeEach
+      fun setUp() = runTest {
+        stubMigrationDependencies()
+        migrationService.startMigration(ContactPersonProfileDetailsMigrationFilter())
+          .also {
+            migrationId = it.migrationId
+            waitUntilCompleted()
+          }
+      }
```

Comment:
I haven't implemented the API yet - will get more tests in for the migration service then add all the endpoints required.